src/main/java/io/scif/formats/ICSFormat.java

```diff
@@ -238,7 +238,13 @@ else if (axis.startsWith("f")) {
 					}
 
 					CalibratedAxis newAxis = FormatTools.createAxis(type);
-					newAxis.setUnit(units == null ? "unknown" : units[n]);
+					if (units == null || n < units.length) {
```

Comment:
This does the wrong thing currently. The comparison needs to be:

    if (units == null || n >= units.length) {

I tested it with this corrected expression, and that fixes loading for my datasets (ics/ids stacks saved from ImageJ via Bio-Formats Exporter).

Reply:
Thank you for catching that.

Reply:
> Can we use any of the datasets that Bio-Formats use for testing?
I am working on a more comprehensive test suite for the new scifio, where each format is tested with example images. Finding appropriately licensed data sets is a challenge though.

Reply:
If needed, I can provide sample datasets for ICS/IDS (written from SVI Huygens, as well as from Bio-Formats within Fiji), Metamorph/Visiview (.nd + .stk) and Zeiss (.czi). I guess they have to be "as small as possible" while still representing a typical dataset layout, right?

Reply:
This would be awesome :)
> I guess they have to be "as small as possible" while still representing a typical dataset layout, right?
Exactly :+1:

Reply:
Thanks @imagejan. In general, we want to build up a ___public domain___ collection of images in various formats, for use as SCIFIO's data-driven test suite. We do not want to commit these to git, but rather download them on demand—probably using SciJava's new `DownloadService`—so that unit tests do not need network access on subsequent invocations, and so that the SCIFIO repositories are not bloated with binary data. (@gab1one and I debated whether to use Git LFS for this, but ultimately figured it should not be necessary.)